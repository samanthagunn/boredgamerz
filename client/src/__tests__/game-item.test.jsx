import React, { useState as useStateMock } from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import GameItem from "../components/game-item";
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: jest.fn(),
}));
const setState = jest.fn()
describe("The game item component...", () => {
  beforeAll(() => {
    useStateMock.mockImplementation((init) => [init, setState]);
  });
  test("will render without crashing.", () => {
      const { baseElement } = render(<GameItem />)
      expect(baseElement).toBeDefined();
  })
  test("will render display description when clicked.", () => {
    const { getByText } = render(<GameItem />)
    fireEvent.click(getByText("Name:"))
    expect(setState).toHaveBeenCalledTimes(1)
})
test("will match snapshot.", () => {
    const tree = renderer.create(<GameItem />).toJSON();
    expect(tree).toMatchSnapshot();
})
});
