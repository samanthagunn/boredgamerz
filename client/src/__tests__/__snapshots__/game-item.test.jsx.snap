// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The game item component... will match snapshot. 1`] = `
<ion-item
  onClick={[Function]}
>
  <ion-card
    onClick={[Function]}
  >
    <ion-card-header>
      <ion-title>
        Name: 
      </ion-title>
      <ion-card-subtitle>
        Game: 
      </ion-card-subtitle>
      <ion-card-subtitle>
        Host: 
      </ion-card-subtitle>
    </ion-card-header>
    <ion-card-content>
      <h2>
        Open Seats: 
      </h2>
      <h3>
        Date: 
         
      </h3>
      <h4>
        Location: 
      </h4>
    </ion-card-content>
  </ion-card>
</ion-item>
`;
